### DISCUSSION TITLE
Vercel will not deploy to my github account -- where to configure the email address vercel pushes are associated with?

### DISCUSSION THREAD

**Discussion by twilwa** [vercel/community#3760] [Help]
### Question

I've been poring over docs trying to figure this out, but the closest I can find is https://vercel.com/guides/why-aren-t-commits-triggering-deployments-on-vercel -- which indicates I should set the email address locally. I've configured it to the anon email address in both windows and WSL with the --global flag, but I'm still unable to deploy. What am I missing?

**Comment by amyegan** [comment]
Hey @twilwa. Did you [create a new project](https://vercel.com/docs/getting-started-with-vercel/import#) and connect your GitHub repo to it? If you walk me through the steps you already tried, then I might be able to recognize what's missing 🙏

**Comment by twilwa** [comment]
I was trying to deploy straight from template after adding a new project, that's where i was runnning into the error. overview>add new>browse all templates>blog with native content editing>deploy>scope to my user account>create>

#461 -> POST /api/v1/integrations/push-to-repo?teamId=team_ujLzpLiAVDp0Dm5IM82F8pp5 pdx1:pdx1:sfo1::8gtng-1692933843637-067ab2227141 (404, Error: remote: error: GH007: Your push would publish a private email address.        
remote: You can make your email public or disable this protection by visiting:        
remote: http://github.com/settings/emails        
To https://github.com/twilwa/blog-nextjs-sanity3
 ! [remote rejected] main -> main (push declined due to email privacy restrictions)

**Reply by victorlora** [reply]
The following worked for me:
1. GitHub > Settings
2. Emails
3. Uncheck "Block command line pushes that expose my email "

Obviously this makes it so the commit author may have your private email address. I ended up creating a private email address and using that for GitHub instead. I use iCloud Hide My Email, but there are other services out there.

Good luck!

Aside: Vercel team, there should be a setting to allow users to set an email address to be used for commit history. When using the private email feature, GitHub provides a "username@users.noreply.github.com" that can be set in the `git` config `user.email`; however, Vercel doesn't allow setting this.

**Comment by twilwa** [comment]
any update on this, vercel team? product remains unusable for me, and i imagine anyone else who doesn't want to reveal their email on github, until resolved.

**Reply by uerkw** [reply]
This is pertinent and necessary for me as well, Vercel team.
